### ISSUE TITLE
POST_COMMANDS_FAILURE are not executed when PRE_COMMANDS fails

### ISSUE THREAD

**Issue by 0x0aNL** [djmaze/resticker#91]
If executing `PRE_COMMANDS` fails (for whatever reason), resticker will not attempt to run restic, nor execute `POST_COMMANDS_FAILURE` (e.g. sending failure notification).

Perhaps it makes more sense to have a separate variable (`PRE_COMMANDS_FAILURE` ?), but in my case just executing `POST_COMMANDS_FAILURE` is useful enough. I haven't made a PR because my bash skills aren't fantastic, but here's how I "fixed" it:

[Comparison](https://github.com/djmaze/resticker/compare/master...0x0aNL:dont-skip-failure-commands)

```
set +e
run_commands "${PRE_COMMANDS:-}"
if [ $? -ne 0 ]
then
    set -e
    run_commands "${POST_COMMANDS_FAILURE:-}"
    exit
else
    set -e
fi
```

In other words, same approach as with `restic backup` itself.

This also applies to the `prune` script.

Thanks a bunch for this great image!

**Comment by djmaze** [comment]
Sorry for the late response. Having a `PRE_COMMANDS_FAILURE` command makes total sense to me. 

Not sure when I have time do this. Would take a PR.

**Comment by 0x0aNL** [comment]
> Sorry for the late response. Having a `PRE_COMMANDS_FAILURE` command makes total sense to me.
> 
> Not sure when I have time do this. Would take a PR.

No worries: would make a PR, but not sure when I have time to do that either. Keeping stale bot at bay until that day ;)
